### ISSUE TITLE
Order of `stop_when_too_toxic()` resulting in unintended results

### ISSUE THREAD

**Issue by DSladeAZ** [brockk/escalation#4]
The order of behaviours added to a model can be important. As an example in the case of `stop_when_n_at_dose()` and `stop_when_too_toxic()` , the latter needs to be above the former if a dose is to be recommended when `stop_when_n_at_dose()` is triggered. 

Correct behaviour:
```
skeleton <- c(0.05, 0.15, 0.30, 0.45)
target <- 0.25
alpha_mean <- 0
alpha_sd <- 1
beta_mean <- 0
beta_sd <- 1

model <- get_trialr_crm(skeleton = skeleton, target = target, model = 'logistic2',  alpha_mean = alpha_mean, alpha_sd = alpha_sd, beta_mean = beta_mean, beta_sd = beta_sd) %>%
  stop_when_too_toxic(dose = 1, tox_threshold = 0.25, confidence = 0.8) %>%
  stop_when_n_at_dose(9, 'recommended') %>%
  stop_at_n(n = 30)

model %>% fit('1NNT 1NNT 1NNT')

The model targets a toxicity level of 0.25.
The model advocates stopping and recommending dose 1.
```

Incorrect behaviour:
```
skeleton <- c(0.05, 0.15, 0.30, 0.45)
target <- 0.25
alpha_mean <- 0
alpha_sd <- 1
beta_mean <- 0
beta_sd <- 1

model <- get_trialr_crm(skeleton = skeleton, target = target, model = 'logistic2',  alpha_mean = alpha_mean, alpha_sd = alpha_sd, beta_mean = beta_mean, beta_sd = beta_sd) %>%
  stop_when_n_at_dose(9, 'recommended') %>%
  stop_when_too_toxic(dose = 1, tox_threshold = 0.25, confidence = 0.8) %>%
  stop_at_n(n = 30)

model %>% fit('1NNT 1NNT 1NNT')

The model targets a toxicity level of 0.25.
The model advocates stopping and recommending no dose.
```

The occurs due to  `stop_when_too_toxic()` always returning `NA`  for `recommended_dose()` if the parent has `continue() == FALSE`.

**Comment by brockk** [comment]
I have fixed the dose recommendation of continuation parameters in 0.1.6. Annoyingly print() is not quite in sync so leaving this issue open.

**Comment by brockk** [comment]
Fixed in 0.1.7
